codegen/templates_test.go

```diff
@@ -25,7 +25,7 @@ func TestClean(t *testing.T) {
 		},
 		{
 			input:    "{role_name}",
-			expected: "rolename",
+			expected: "roleName",
```

Comment:
Can you add some additional tests around camel casing? I'm thinking less happy-path tests like:
`{role_name_}`
`{_role_name}`
`{role__name}`
`{role_name_here}`
`{rOlE_nAmE}`
`{ROLE_NAME}`

Reply:
Sure, I added tests for all of those but one. https://github.com/terraform-providers/terraform-provider-vault/pull/749/commits/8c7130421beaa54b1d6f9be2081e3cd3521b8959

The fields being passed in here originate from the OpenAPI doc and they all look like this:
```
"/transit/hash/{urlalgorithm}"
"/transit/hmac/{name}"
"/transit/hmac/{name}/{urlalgorithm}"
"/transit/keys"
"/transit/keys/{name}"
"/transit/keys/{name}/config"
"/transit/keys/{name}/rotate"
"/transit/keys/{name}/trim"
"/transit/random"
"/transit/random/{urlbytes}"
"/transit/restore"
"/transit/restore/{name}"
"/transit/rewrap/{name}"
"/transit/sign/{name}"
"/transit/sign/{name}/{urlalgorithm}"
"/transit/verify/{name}"
"/transit/verify/{name}/{urlalgorithm}"
```
So, the algorithm and tests are more than robust enough for this. If ever there were a super unexpected input, that would be caught during testing the resource and the algorithm would be updated. It's hard to imagine, though, that we'd have an endpoint with a path parameter that started with an `_`, though, so I felt comfortable not accounting for that.